core/che-core-api-jdbc/src/main/java/org/eclipse/che/api/core/jdbc/jpa/event/CascadeRemovalEvent.java

```diff
@@ -0,0 +1,29 @@
+/*******************************************************************************
+ * Copyright (c) 2012-2016 Codenvy, S.A.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *
+ * Contributors:
+ *   Codenvy, S.A. - initial API and implementation
+ *******************************************************************************/
+package org.eclipse.che.api.core.jdbc.jpa.event;
+
+/**
+ * Special event type which is needed only for
+ * notification in the process of cascade removing.
+ *
+ * @author Anton Korneta
+ */
+public abstract class CascadeRemovalEvent {
+    private final RemovalContext context;
```

Comment:
What do you think about using direct field initialization in this way you can get rid of constructor.
```java
private final RemovalContext context = new RemovalContext();
```